src/vm/gcinfodecoder.cpp

```diff
@@ -1539,8 +1539,7 @@ void GcInfoDecoder::ReportRegisterToGC(  // AMD64
     LOG((LF_GCROOTS, LL_INFO1000, "Reporting " FMT_REG, regNum ));
 
     OBJECTREF* pObjRef = GetRegisterSlot( regNum, pRD );
-
-#ifdef FEATURE_PAL
+#if defined(FEATURE_PAL) && !defined(SOS_TARGET_AMD64) 
```

Comment:
The SOS_TARGET_AMD64 doesn't seem like the right macro to use. We will want the extension to work for any target eventually. Why haven't you used the DACCESS_COMPILE?